include/extractor/scripting_environment.hpp

```diff
@@ -23,18 +26,28 @@ namespace extractor
 class ScriptingEnvironment
 {
   public:
+    struct Context
+    {
+        Context();
+        ~Context();
+
+        ProfileProperties properties;
+        SourceContainer sources;
+        lua_State *state;
+    };
+
     explicit ScriptingEnvironment(const std::string &file_name);
 
     ScriptingEnvironment(const ScriptingEnvironment &) = delete;
     ScriptingEnvironment &operator=(const ScriptingEnvironment &) = delete;
 
-    lua_State *GetLuaState();
+    Context &GetContex();
```

Comment:
Yes should be a unique_ptr actually. Function can't be const because this modifies the locks.